docs/build.md

```diff
@@ -0,0 +1,60 @@
+## Build and package configuration
+
+Build variables should be defined in a single file,
+so that anyone who wants to build Rancher needs to only edit this file to change configuration and dependency versions.
+
+Rancher relies on various subcomponents, such as the webhook.
+These typically need to have set versions for Rancher to build and run properly.
+Build variables can be used in different places and supplied to the applications in a variety of ways,
+including as environment variables in Dockerfiles, constants in Go code, and so on.
+
+The [build.yaml](../build.yaml) file is the single source of truth. It lists all values by name and value.
+Changes to it should be committed to source control.
+
+### Update an existing value
+
+Edit the [build.yaml](../build.yaml) file and update the desired value. Run `go generate`. Commit any changes to source
+control.
+Then re-build Rancher with `make build` or re-package it with `make package`.
+
+### Add a new value
+
+Pass a configuration value in one of two ways:
```

Comment:
The current wording sounds like the user needs to perform steps 1 or 2.
We should update the wording to explain this is what the build scripts are doing.
```suggestion
The build configuration passes a configuration value in one of two ways:
```

Reply:
No, this is indeed for devs who want to add a new value to the build-release system. This is a one-time operation. For example, you want to add support for the charts branch - you follow these instructions.

Reply:
> you follow these instructions.

Your numbered points are not instructions they are explaining what the build configuration is doing. 

`Pass a configuration value in one of two ways:` Is implying that we are going to tell the user how to pass the configuration value. But the first step says `This is not something the dev is doing.`

I understand what you are trying to say because I am familiar with the process but for a new dev they would expect the numbered sections to be steps to perform instead you are using them as section headers.

> 1. As a `--build-arg` to `docker image build`. Capture it as an `ARG` in the Dockerfile
   via `docker build --build-arg MYVAR="$MYVAR" ...`
 As a reference to a constant found in the generated Go [file](../pkg/buildconfig/constants.go).
This is not something the dev is doing. 
> 2. As a reference to a constant found in the generated Go [file](../pkg/buildconfig/constants.go).

Reply:
👀